### ISSUE TITLE
@DistinctUntilChanged doesn't clear cache in TiFragment after orientation change

### ISSUE THREAD

**Issue by passsy** [grandcentrix/ThirtyInch#29]
Reproduce:
1. call `showGallery(new ArrayList())` TiFragment implementing this view interface method

``` java
    @DistinctUntilChanged
    void showGallery(List<Image> images);
```
1. make orientation change
2. call view again with same argument (`new ArrayList()`)

_Expected behavior:_
View method gets called

_Actual behavior:_
View method doesn't get called because it's the same data. This is wrong because `onCreateView()` was called and a new view implementation was created.

Workaround:

clear the cache manually in `onCreateView()`

``` java
final DistinctUntilChangedInterceptor distinctUntilChangedInterceptor = (DistinctUntilChangedInterceptor)
                getInterceptors(interceptor -> interceptor instanceof DistinctUntilChangedInterceptor).get(0);
        distinctUntilChangedInterceptor.clearCache(this);
```

**Comment by rberghegger** [comment]
I tried to reproduce this issue and noticed that the bug does seem to be fixed.
Please notice that it is very important to call `super.onCreateView()` in your `onCreateView()` implementation because otherwise `TiFragment` is not able to invalidate the cache of the latest bound view itself.

I would suggest to close this issue and look into my pull request which restricts the `TiFragment` api to subclasses and enforces the call of the super implementation when overriding important lifecycle methods.

**Comment by StefMa** [comment]
I can reproduce the error on "not calling super" as @rberghegger mentioned.

So I agree with @rberghegger.
Closed and see #79
